src/main/java/org/myrobotlab/service/ProgramAB.java

```diff
@@ -1361,6 +1349,16 @@ public void onUtterance(Utterance utterance) throws Exception {
         if (!config.sleep) {
           shouldIRespond = true;
         }
+        
+        if (config.channels != null && config.channels.size() > 0) {
+          // assume false
+          shouldIRespond = false;
+          for (String channelName : config.channels) {

```

Comment:
where as this is functional, it's faster/more scalable/easier to use a set instead of a list for the channels here to determine if the bot should respond to a particular channel, maybe just add a comment in the code that is a TODO to refactor this logic to avoid the loop here and use a set instead.